### PR TITLE
fix: EXPERIMENTAL_protocol_config applies overrides from EpochConfig

### LINKED ISSUE #9553: Wrong genesis config in EXPEIMENTAL_protocol_config
```
% http post https://rpc.testnet.near.org jsonrpc=2.0 id=dontcare method=EXPERIMENTAL_protocol_config params:='{ "finality": "final" }' | jq '.result.num_block_producer_seats'
200
```

But the value is actually `100`, not `200`:
https://github.com/near/nearcore/blob/a5238cc70e268fe6da218c8367b053d53711ac6f/core/primitives/src/epoch_manager.rs#L134

### PR DESCRIPTION
**Pull request by nikurt**
Testing this PR on a testnet node, the diff with results from an rpc node is the following. Both changes look good to me.

```
10c10
<   "chunk_producer_kickout_threshold": 90,
---
>   "chunk_producer_kickout_threshold": 80,
28c28
<   "num_block_producer_seats": 200,
---
>   "num_block_producer_seats": 100,
```

Proofs:
* https://github.com/near/nearcore/blob/a5238cc70e268fe6da218c8367b053d53711ac6f/core/primitives/src/epoch_manager.rs#L134
* https://github.com/near/nearcore/blob/6f324e84a7a7162956f0b9985094ee146919f5ae/core/primitives/src/epoch_manager.rs#L149

Fix #9553 

Also adds new fields to the response:
* shard_layout
* num_chunk_only_producer_seats
* minimum_validators_per_shard
* minimum_stake_ratio
* max_kickout_stake_perc

### CODE REVIEW

**Review by wacban — DISMISSED** [review]
Please rebase or merge first. 
LGTM

File: core/primitives/src/runtime/config.rs [hunk 1]
```diff
@@ -39,10 +39,9 @@ impl RuntimeConfig {
 
     pub fn test() -> Self {
         let config_store = super::config_store::RuntimeConfigStore::new(None);
-        let mut wasm_config = near_vm_runner::logic::Config::clone(
```
[the remark this answers is not in the document]
Comment (nikurt) [reply core/primitives/src/runtime/config.rs]: Yes, resolved by rebasing.

**Comment by nikurt** [comment]
Thanks @wacban 
Rebased, added a TODO, please take a look.

**Review by wacban — APPROVED** [review]
Thanks!

**Pull request merged**